### PR TITLE
[PATCH v3] linux-gen: update ARM HWCAP2 flags

### PR DESCRIPTION
**Pull request by psavol**
Add missing HWCAP2 flags and simplify the code checking the flags.

### CODE REVIEW

File: platform/linux-generic/arch/aarch64/cpu_flags.c [hunk 1]
```diff
@@ -23,6 +23,10 @@ typedef struct {
 	bool valid;
 } hwcap_feat_flag_t;
 
+/* Linux HWCAP and HWCAP2 flags
+ *
+ * See https://docs.kernel.org/arm64/elf_hwcaps.html for meaning of each flag.
```
Comment (TuomasTaipale) [inline platform/linux-generic/arch/aarch64/cpu_flags.c:27]: "/arch/" seems to be missing (https://docs.kernel.org/arch/arm64/elf_hwcaps.html).

File: platform/linux-generic/arch/aarch64/cpu_flags.c [hunk 2]
```diff
@@ -437,7 +437,182 @@ static hwcap_feat_flag_t hwcap2_flags[] = {
...
+		.bit_mask = HWCAP2_SME_B16B16,
+#endif
+	},
+
+	{
+		.name = "SME_F16F16",
+#ifdef HWCAP2_SME_F16F16
+		.bit_mask = HWCAP2_SME_F16F16,
+#endif
+	},
```
Comment (TuomasTaipale) [inline platform/linux-generic/arch/aarch64/cpu_flags.c:608]: These four appear to be without the "_" after "SME" in the kernel documentation, e.g. `HWCAP2_SME_F16F16` vs. `HWCAP2_SMEF16F16`.

Comment (psavol) [reply platform/linux-generic/arch/aarch64/cpu_flags.c:608]: Seems that kernel documentation is not in sync with the header file:

https://github.com/torvalds/linux/blob/master/arch/arm64/include/uapi/asm/hwcap.h

#define HWCAP2_SME_I16I32	(1UL << 39)
#define HWCAP2_SME_BI32I32	(1UL << 40)
#define HWCAP2_SME_B16B16	(1UL << 41)
#define HWCAP2_SME_F16F16	(1UL << 42)
#define HWCAP2_MOPS		(1UL << 43)

Comment (TuomasTaipale) [reply platform/linux-generic/arch/aarch64/cpu_flags.c:608]: OK, a bit unfortunate for the documentation reader.

File: platform/linux-generic/arch/aarch64/cpu_flags.c (hunk 1 above)
Comment (psavol) [reply platform/linux-generic/arch/aarch64/cpu_flags.c:27]: Fixed in v2.

File: platform/linux-generic/arch/aarch64/cpu_flags.c [hunk 3]
```diff
@@ -437,7 +437,182 @@ static hwcap_feat_flag_t hwcap2_flags[] = {
 #ifdef HWCAP2_MTE
```
[the remark this answers is not in the document]
Comment (psavol) [reply platform/linux-generic/arch/aarch64/cpu_flags.c:437]: Year updated in v2

**Review by TuomasTaipale — APPROVED** [review]

**Pull request merged**